### ISSUE TITLE
Does not work on windows

### ISSUE THREAD

**Issue by 89z** [wader/fq#21]
~~~
PS C:\fq-master> .\fq.exe
src:1:0: compile: open @builtin\internal.jq: file does not exist
~~~

**Comment by wader** [comment]
Hi, thanks for the report. Sorry that windows testing is lacking a bit a the moment.

If possible you can try change https://github.com/wader/fq/blob/master/pkg/interp/interp.go#L869 to `return strings.Join([]string{base, path}, "/")`?  I think the problem is that go embed always uses "/" no matter what platforms it runs on. Will think about a more proper solution if that is the case.

Hope i can get a more proper windows dev and CI setup soon.

**Comment by wvlover15** [comment]
> Hi, thanks for the report. Sorry that windows testing is lacking a bit a the moment.
> 
> If possible you can try change https://github.com/wader/fq/blob/master/pkg/interp/interp.go#L869 to `return strings.Join([]string{base, path}, "/")`? I think the problem is that go embed always uses "/" no matter what platforms it runs on. Will think about a more proper solution if that is the case.
> 
> Hope i can get a more proper windows dev and CI setup soon.

Could you please build that fix for us for windows?

**Comment by wader** [comment]
Sure can do. Just wanted someone to test it. Could you also try run all tests to make sure it does not break something else? ”go test ./pkg/interp ./format” or ”make test” if you have make installed.

**Comment by wvlover15** [comment]
> Sure can do. Just wanted someone to test it. Could you also try run all tests to make sure it does not break something else? ”go test ./pkg/interp ./format” or ”make test” if you have make installed.

Alright. I'll test with make

**Comment by kiwijam** [comment]
That does seem to work. Thanks!

**Comment by zackmark29** [comment]
> Sure can do. Just wanted someone to test it. Could you also try run all tests to make sure it does not break something else? ”go test ./pkg/interp ./format” or ”make test” if you have make installed.

@kiwijam already confirmed it is the fix. I'll just wait for the binary :)

**Comment by wader** [comment]
Thanks for testing. Still want to think a bit if there is some nicer way to solve this and also how to add a regression test to not break this again.

**Comment by wader** [comment]
Finally had some time setup a windows dev env. It's a bit tricky to fix as it affects relative path includes

**Comment by wader** [comment]
Sorry for lack update, im out traveling atm so dont have much time. Part of fixing this is also to fix some tests that are broken on windows, probably not becuse of bugs, but dont know yet.

**Comment by razor-87** [comment]
> 
> If possible you can try change https://github.com/wader/fq/blob/master/pkg/interp/interp.go#L869 to `return strings.Join([]string{base, path}, "/")`? I think the problem is that go embed always uses "/" no matter what platforms it runs on. Will think about a more proper solution if that is the case.
> 

Thanks, now it works, only the line number is different: https://github.com/wader/fq/blob/6fbb8563a9f67a68a9f8c4cd29221b3b80489a0f/pkg/interp/interp.go#L787

**Comment by wader** [comment]
Hi, now master should work on windows, also tests, and there is a CI build.

Can someone more test just to be sure? my testing is via windows vm and github action host so should be fine but who knows.

**Comment by wader** [comment]
Fixed by #52 and is included in v0.0.3
